cmd/gateway-main.go

```diff
@@ -185,11 +185,24 @@ func gatewayMain(ctx *cli.Context) {
 	}
 
 	// First argument is selected backend type.
-	backendType := ctx.Args().First()
+	backendType := ctx.Args().Get(0)
+	// Second argument is the endpoint address (optional)
+	endpointAddr := ctx.Args().Get(1)
+	// Third argument is the address flag
+	serverAddr := ctx.String("address")
```

Comment:
As this PR is doing input args validation, I see below things here.
* non-empty `serverAddr` should be checked with `CheckLocalServerAddr(serverAddr string)`
* Here valid value format of `endpointAddr` is "", "remote-server", "remote-server:port", "http://remote-server", "https://remote-server:port".  Other than these values are error case.
* there is a special case where `serverAddr` can be a local host only for testing.  In this case, port in `serverAddr` should not be same as `endpointAddr` (as this PR addresses).

If the PR scope is only to last point only, I am not sure how critical to fix that case.  Additionally not sure why gateway is not failing to return error on port already in use.